ImageClassification/CIFAR10/DeepResNet/resnet_eval.py

```diff
@@ -15,24 +15,50 @@
 # ----------------------------------------------------------------------------
 import os
 
+import numpy as np
 from neon.util.argparser import NeonArgparser
 from neon.util.persist import load_obj
 from neon.transforms import Misclassification, CrossEntropyMulti
 from neon.optimizers import GradientDescentMomentum
 from neon.layers import GeneralizedCost
 from neon.models import Model
-from neon.data import DataLoader, ImageParams
+
+from neon.data.dataloader_transformers import OneHot, TypeCast, BGRMeanSubtract
+from neon.data.aeon_shim import AeonDataLoader
 
 # parse the command line arguments (generates the backend)
 parser = NeonArgparser(__doc__)
 args = parser.parse_args()
 
-# setup data provider
-test_dir = os.path.join(args.data_dir, 'val')
-shape = dict(channel_count=3, height=32, width=32)
-test_params = ImageParams(center=True, flip=False, **shape)
-common = dict(target_size=1, nclasses=10)
-test_set = DataLoader(set_name='val', repo_dir=test_dir, media_params=test_params, **common)
+def wrap_dataloader(dl, dtype=np.float32):
+    dl = OneHot(dl, index=1, nclasses=10)
+    dl = TypeCast(dl, index=0, dtype=dtype)
+    dl = BGRMeanSubtract(dl, index=0)
+    return dl
+
+def config(manifest_filename, manifest_root, batch_size, subset_pct):
+    image_config = {"type": "image",
+                    "height": 32,
+                    "width": 32}
+    label_config = {"type": "label",
+                    "binary": False}
+    augmentation = {"type": "image",
+                    "crop_enable": True,
+                    "center": True,
+                    "flip_enable": False}
+
+    return {'manifest_filename': manifest_filename,
+            'manifest_root': manifest_root,
+            'batch_size': batch_size,
+            'subset_fraction': float(subset_pct/100.0),
+            'etl': [image_config, label_config],
+            'augmentation': [augmentation]}
+
+def make_val_config(manifest_filename, manifest_root, batch_size, subset_pct=100):
+    val_config = config(manifest_filename, manifest_root, batch_size, subset_pct)
+    return wrap_dataloader(AeonDataLoader(val_config))
+
+test_set = make_val_config(args.manifest["val"], args.manifest_root, batch_size=args.batch_size)
 
 model = Model(load_obj(args.model_file))
```

Comment:
model=Model(load_obj(args.model_file)) will break the code, since neon v2.2 and v2.3 was not supposed to initialize a model using old weights. model=Model(load_obj(args.model_file) works with new weights. 
So how do we get the new format of an old args.model_file. 
Can we try to use model=Model(layers=cifar10_layers). 
And do model.load_params(args.model_files, load_state).. 

When we see there is an assert failure. This usually mean only a subset of the cifar10 layers was involved in saving the weights. So we need to print out those layer information from the old weight file and see which layers are missing. We then delete those layers (temporarily) from cifar10 layers.

Reply:
Is resnet_eval.py still functional/successful in loading weights?